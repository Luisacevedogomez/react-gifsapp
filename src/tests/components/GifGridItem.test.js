import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas al componente GifGridItem", () => {
    const title = "Un titulo";
    const url = "https://localhost/foto.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test("debe mostrar el componente correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe tener un parrafo con el title", () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    });

    test("debe tener una imagen con el url y el title", () => {
      const img = wrapper.find("img");
      expect(img.prop('src')).toBe(url);
      expect(img.prop("alt")).toBe(title);
    });

    test('debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    }) 

});
