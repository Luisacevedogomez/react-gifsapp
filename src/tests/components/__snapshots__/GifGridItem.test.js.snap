// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/foto.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
